hazelcast/src/main/java/com/hazelcast/config/replacer/EncryptionReplacer.java

```diff
@@ -0,0 +1,290 @@
+            }
+            if (passwordUserProperties) {
+                baos.write(System.getProperty("user.home").getBytes(UTF8_CHARSET));
+                baos.write(System.getProperty("user.name").getBytes(UTF8_CHARSET));
+            }
+            if (passwordNetworkInterface != null) {
+                try {
+                    NetworkInterface iface = NetworkInterface.getByName(passwordNetworkInterface);
+                    baos.write(iface.getHardwareAddress());
+                } catch (SocketException e) {
+                    throw rethrow(e);
+                }
+            }
+            return new String(Base64.encode(baos.toByteArray()), UTF8_CHARSET).toCharArray();
+        } catch (Exception e) {
+            throw rethrow(e);
+        }
+    }
+
+    public static final void main(String... args) throws Exception {
+        if (args == null || args.length < 1 || args.length > 2) {
+            System.err.println("Usage:");
+            System.err.println("\tjava -D<propertyName>=<propertyValue>  " + EncryptionReplacer.class.getName()
+                    + " \"<String To Encrypt>\" [iterations]");
+            System.err.println();
+            System.err.println("The replacer configuration can be loaded either from hazelcast/hazelcast-client XML file:");
+            System.err.println("\t-Dhazelcast.config=/path/to/hazelcast.xml");
+            System.err.println();
+            System.err.println("or provided directly via following property names:");
+            System.err.println("\t" + PROPERTY_CIPHER_ALGORITHM);
+            System.err.println("\t" + PROPERTY_KEY_LENGTH_BITS);
+            System.err.println("\t" + PROPERTY_SALT_LENGTH_BYTES);
+            System.err.println("\t" + PROPERTY_SECRET_KEY_ALGORITHM);
+            System.err.println("\t" + PROPERTY_SECRET_KEY_FACTORY_ALGORITHM);
+            System.err.println("\t" + PROPERTY_SECURITY_PROVIDER);
+            System.err.println("\t" + PROPERTY_PASSWORD_FILE);
+            System.err.println("\t" + PROPERTY_PASSWORD_NETWORK_INTERFACE);
+            System.err.println("\t" + PROPERTY_PASSWORD_USER_PROPERTIES);
+            System.err.println();
+            System.err.println("Values available for property " + PROPERTY_PASSWORD_NETWORK_INTERFACE);
+            Enumeration<NetworkInterface> networkInterfaces = NetworkInterface.getNetworkInterfaces();
+            while (networkInterfaces.hasMoreElements()) {
+                NetworkInterface networkInterface = networkInterfaces.nextElement();
+                byte[] hardwareAddress = networkInterface.getHardwareAddress();
+                if (hardwareAddress != null) {
+                    System.err.println("\t" + networkInterface.getName());
+                }
+            }
+            System.err.println();
+            System.exit(1);
+        }
+        System.out.println(encrypt(args));
+    }
+
+    protected static String encrypt(String... args) throws Exception, FileNotFoundException {
```

Comment:
Minor, this `FileNotFoundException` is not needed here, there is a `throws Exception declaration`.